addons/website_sale/static/src/js/editor/snippets.options.js

```diff
@@ -1,8 +1,16 @@
 import options from "@web_editor/js/editor/snippets.options";
 
 options.registry.MegaMenuLayout = options.registry.MegaMenuLayout.extend({
-    _fetchEcommerceCategories() {
-        return this.containerEl.classList.contains('fetchEcomCategories');
+
+    /**
+     * Initialize fetchEcomCategories to the value of the button.
+     *
+     * @private
+     * @returns {string} xmlid of the current template.
+     */
+    start() {
+        this._super();
+        this.fetchEcomCategories = this.containerEl.classList.contains('fetchEcomCategories');
```

Comment:
Can't we do this with a dataset?
```suggestion
        this.fetchEcomCategories = !!this.containerEl.dataset.fetchEcomCategories;
```

Reply:
After further investigation, it turns out it's more difficult to do it with a dataset as it requires more overrides to fully support it in the web editor, and it seems like the dataset wasn't saved on the container element when saving the page.

Therefore, we decided to stick with the original solution, using a class to determine whether the feature is enabled or not.